### ISSUE TITLE
Reading log entry shows blank title/author if the logged work has been merged into another work

### ISSUE THREAD

**Issue by mheiman** [internetarchive/openlibrary#5551]
### Evidence / Screenshot (if possible)
<img width="823" alt="Screen Shot 2021-08-16 at 4 26 39 PM" src="https://user-images.githubusercontent.com/886889/129631687-9fdea28d-1be0-4f08-a137-54354f9ea798.png">

### Steps to Reproduce

1. Add a work to your reading log
2. Merge the work you added into another work so that the original work becomes a redirect
3. View the reading log

* Actual: The merged work has no author/title/cover
* Expected: A proper work listing

### Details

- **Logged in (Y/N)?** Y
- **Browser type/version?** N/A
- **Operating system?** N/A
- **Environment (prod/dev/local)?** prod

### Proposal & Constraints

Modify process_logged_books in /plugins/upstream/account.py to follow redirect entries returned by SOLR instead of just passing them to be rendered.

### Stakeholders

@mheiman @mekarpeles

**Comment by Freso** [comment]
Possibly related and/or duplicate issue: https://github.com/internetarchive/openlibrary/issues/11646

**Comment by cdrini** [comment]
This should no longer be the case; can you verify, @mheiman ? This was fixed when the reading logs were changed to be edition-aware.

**Comment by mheiman** [comment]
Yes, the unknown author issue is resolved. The reading log data is still bad, which is a pain if you're using the API, but that's a different problem.

**Comment by cdrini** [comment]
Sweet! Would you mind opening a new issue for the reading log with some examples? That should make it easier to for someone to tackle that specific issue.
